server/apps/immich/src/api-v1/user/user.service.ts

```diff
@@ -70,142 +61,50 @@ export class UserService {
   }
 
   async createUser(createUserDto: CreateUserDto): Promise<UserResponseDto> {
-    const user = await this.userRepository.getByEmail(createUserDto.email);
-
-    if (user) {
-      throw new BadRequestException('User exists');
-    }
-
-    try {
-      const savedUser = await this.userRepository.create(createUserDto);
-
-      return mapUser(savedUser);
-    } catch (e) {
-      Logger.error(e, 'Create new user');
-      throw new InternalServerErrorException('Failed to register new user');
-    }
+    const createdUser = await this.userCore.createUser(createUserDto);
+    return mapUser(createdUser);
   }
 
   async updateUser(authUser: AuthUserDto, updateUserDto: UpdateUserDto): Promise<UserResponseDto> {
-    const requestor = await this.userRepository.get(authUser.id);
-
-    if (!requestor) {
-      throw new NotFoundException('Requestor not found');
-    }
-
-    if (!requestor.isAdmin) {
-      if (requestor.id !== updateUserDto.id) {
-        throw new BadRequestException('Unauthorized');
-      }
-    }
-
-    const user = await this.userRepository.get(updateUserDto.id);
+    const user = await this.userCore.get(updateUserDto.id);
     if (!user) {
       throw new NotFoundException('User not found');
     }
-    try {
-      user.password = updateUserDto.password ?? user.password;
-      user.firstName = updateUserDto.firstName ?? user.firstName;
-      user.lastName = updateUserDto.lastName ?? user.lastName;
-      user.isAdmin = updateUserDto.isAdmin ?? user.isAdmin;
-      user.shouldChangePassword = updateUserDto.shouldChangePassword ?? user.shouldChangePassword;
-      user.profileImagePath = updateUserDto.profileImagePath ?? user.profileImagePath;
```

Comment:
These didn't get added anywhere as far as I can tell, so I think there's a problem here. Basically, `userRepository.update()` is implemented with `repository.save()`, so it actually requires the full entity. Fields that are omitted during a "save" are removed, which is why these all are listed here. If you aren't going include these lines (merge the dto into the user somewhere and pass the user to repositroy.update), you should update the repository update implementation to not use save.

Reply:
Ah right, that's a good catch, I didn't see that. Why is it doing a save instead of un update 🤔. I trusted the interface of the userRepository that takes a partial entity, so that is not right if is doing a save.
```typescript
update(id: string, user: Partial<UserEntity>): Promise<UserEntity>;
```

Actually I can see in the history that interface was changed a month ago. Before it was:
```typescript
update(user: UserEntity, updateUserDto: UpdateUserDto): Promise<UserEntity>;
```
which works for updating an entity with `save`

I prefer the new interface though, it makes more sense to me. What do you think about changing the repository implementation to use `update` instead of `save`?

Reply:
That's fine, but I don't think cascades or relations are saved that way. Probably not an issue yet. Also update doesn't return a full user, so you'd need to requery it afterwards. I do like the interface of "only pass what you want to change". So either query the user, merge the new properties and return the whole thing. Or, do an update, then query the new user and return it. Or, I guess change the interface to not return a user entity.

This is a thing we should decide and then make consistent across all the repos though.

Reply:
Ah I think I misinterpreted what update you were referring to. 

I like the interface for an update not requiring passing the full entity, but only the fields that have changed. 

I'm fine if we want to always require the full entity though and just do a save on it, but I don't think the repository should be the one merging the two objects together. Just pass the already merged object to the repo and all it does is save it.

Reply:
I was checking the other repositories and the interfaces in the UserRepository for create and update before they were changed.
This was create:
```typescript
create(createUserDto: CreateUserDto): Promise<UserEntity> 
```
And this was update:
```typescript
update(user: UserEntity, updateUserDto: UpdateUserDto): Promise<UserEntity> 
```
Originally the repository was designed to be responsible for doing any work in the Entities (updating or creating). When an entity is returned from the repository it shouldn't be mutated by the services, and the services shouldn't create Entities.

- For create we pass in the createDto and the repository uses it to do an insert in the DB and returns an entity (since we are using ORM it actually creates a new Entity object and uses it to insert it. Then it returns a new created Entity)
- For update we pass in the entity that we got from a get (without mutating) and the UpdateDto. Then the repository updates it in the DB (since we are using ORM it actually merges the DTO into the Entity)

However, looking the signature of `save`, it looks like you don't need to pass the full entity, a partial would be fine, so we don't need to do the merge at all, which means we can keep following the same "original" rules and make repositories less restricted and also have less logic
```typescript
Repository<UserEntity>.save<Partial<UserEntity>>(entity: Partial<UserEntity>, options?: SaveOptions | undefined): Promise<Partial<UserEntity> & UserEntity> 
```
Which means that the current signature and implementation should be fine.
And actually I just remember I did some manual testing updating user details with the current logic and it was working fine

I think we can leave it as it is, and agree if we should use this same kind of signatures for the other repositories to make it consistent. In my opinion, it makes sense. Create and update just need `Partial<Entity>`